### PR TITLE
add search model group permission to ml_read_access role

### PR DESCRIPTION
**Pull request by rbhavna**
### Description
[Describe what this change achieves]
* Category (Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation)
Enhancement

* Why these changes are required?
ML-commons has a new index "model group" added in 2.8 release. This PR adds the search model group permission to the "ml_read_access" role.

* What is the old behavior before changes and new behavior after changes?
We want a user with "ml_read_access" role to be able to search the model group index.  Before this change, user is not given that permission which is contrary to the expected behaviour. 
This change allows any user with "ml_read_access" role to search the model group index.

### Issues Resolved

Is this a backport? If so, please add backport PR # and/or commits #

### Testing
[Please provide details of testing done: unit testing, integration testing and manual testing]

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: config/roles.yml [hunk 1]
```diff
@@ -262,6 +262,7 @@ ml_read_access:
     - 'cluster:admin/opensearch/ml/models/search'
     - 'cluster:admin/opensearch/ml/tasks/get'
     - 'cluster:admin/opensearch/ml/tasks/search'
+    - 'cluster:admin/opensearch/ml/model_groups/search'
```
Comment (peternied) [inline config/roles.yml:265]: Could you add this action in alphabetic order please?

Comment (rbhavna) [reply config/roles.yml:265]: Could you please suggest if this line should go at line 261 or 263?

**Review by peternied — APPROVED** [review]

**Comment by peternied** [comment]
@rbhavna Thanks - is there a specific version you'd like this change to be back ported into?

File: config/roles.yml (hunk 1 above)
Comment (rbhavna) [reply config/roles.yml]: > @rbhavna Thanks - is there a specific version you'd like this change to be back ported into?

Yes, to 2.x and 2.8

**Comment by rbhavna** [comment]
> @rbhavna Thanks - is there a specific version you'd like this change to be back ported into?

Yes, to 2.x and 2.8

**Review by RyanL1997 — APPROVED** [review]

**Comment by peternied** [comment]
2.8 has already released, but we will make sure this is merged for the 2.9 release.

**Comment by rbhavna** [comment]
@peternied @RyanL1997 Thanks

**Pull request merged**